src/datachain/lib/listing.py

```diff
@@ -97,15 +109,68 @@ async def _scandir(client, prefix, fetch_workers) -> AsyncIterator:
     await main_task
 
 
-def list_bucket(uri: str, client_config=None, fetch_workers=FETCH_WORKERS) -> Callable:
+def list_bucket(uri: str, fetch_workers=FETCH_WORKERS, **kwargs) -> Callable:
     """
     Function that returns another generator function that yields File objects
     from bucket where each File represents one bucket entry.
     """
 
     def list_func() -> Iterator[File]:
-        config = client_config or {}
-        client, path = Client.parse_url(uri, None, **config)  # type: ignore[arg-type]
+        client, path = Client.parse_url(uri, None, **kwargs)  # type: ignore[arg-type]
         yield from iter_over_async(_scandir(client, path, fetch_workers), get_loop())
 
     return list_func
+
+
+def ls(
+    dc: "DataChain",
+    path: str,
+    recursive: Optional[bool] = True,
+    object_name="file",
+):
+    """
+    Return files by some path from DataChain instance which contains bucket listing.
+    Path can have globs.
+    If recursive is set to False, only first level children will be returned by
+    specified path
+    """
+
+    def _file_c(name: str) -> Column:
+        return Column(f"{object_name}.{name}")
+
+    dc = dc.filter(_file_c("is_latest") == true())
+
+    if recursive:
+        if not path or path == "/":
+            # root of a bucket, returning all latest files from it
+            return dc
+
+        if not glob.has_magic(os.path.basename(os.path.normpath(path))):
+            # path is not glob, so it's pointing to some directory or a specific
+            # file and we are adding proper filter for it
+            return dc.filter(
+                (_file_c("path") == path)
+                | (_file_c("path").glob(path.rstrip("/") + "/*"))
+            )
+
+        # path has glob syntax so we are returning glob filter
+        return dc.filter(_file_c("path").glob(path))
+    # returning only first level children by path
+    return dc.filter(pathfunc.parent(_file_c("path")) == path.lstrip("/").rstrip("/*"))
```

Comment:
why do we strip only `/*` - can it be a different glob there?

Reply:
should we also return glob filter here even if it's nor recursive?

Reply:
Good question. I was kind of just c/p this ones from the old code so not sure if there was any specific reason to do so but I agree it does look strange. 

I'm not even sure if globs + non recursive even make sense TBO ... and `recursive` naming is also a little bit strange. From what I remember idea here was to return only first level children for some specific path.

Reply:
is there any resolution to this? I'm not sure I completely understand this listing logic in case of globs +/- recursive

Reply:
I tried to remember and I'm pretty sure this was decision way back in the early days of dql where we decided to support just this simple glob when listing/indexing and to add support others in future but we just never added it